ganga/GangaCore/Lib/BackendScriptTemplate.py.template

```diff
@@ -0,0 +1,207 @@
+    if mimetypes.guess_type(f)[1] in ['gzip', 'bzip2']:
+        getPackedInputSandbox(f)
+    else:
+        shutil.copy(f, os.path.join(os.getcwd(), os.path.basename(f)))
+
+# -- END OF MOVED CODE BLOCK
+
+
+#get input files
+###DOWNLOADINPUTFILES###
+
+# create inputdata list
+###CREATEINPUTDATALIST###
+
+gangadir = ###GANGADIR###
+sys.path.insert(0,gangadir)
+sys.path.insert(0,os.path.join(os.getcwd(),PYTHON_DIR))
+
+runenv = os.environ.copy()
+for key,value in environment.items():
+    runenv[key] = value
+
+if Backend == "BATCH":
+    outfile = os.dup(sys.stdout.fileno())
+    errorfile = os.dup(sys.stderr.fileno())
+
+    print("--- GANGA APPLICATION OUTPUT BEGIN ---", file=sys.stdout)
+    print("--- GANGA APPLICATION ERROR BEGIN ---", file=sys.stderr)
+    flush_file(sys.stdout)
+    flush_file(sys.stderr)
+
+elif Backend == "LOCAL":
+    outfile=open('stdout','w')
+    errorfile=open('stderr','w')
+
+sys.stdout=open('./__syslog__','w')
+sys.stderr=sys.stdout
+
+###VIRTUALIZATION###
+
+try:
+    child = subprocess.Popen(execmd, shell=False, stdout=outfile, stderr=errorfile, env=runenv)
+except OSError as x:
+    failurereport(statusfile, 'PROBLEM STARTING THE APPLICATION SCRIPT: \'%s\' \'%s\''%(execmd,str(x)))
+
+if Backend == "LOCAL":
+    print('PID: %d'%child.pid, file=statusfile)
+    flush_file(statusfile)
+
+result = -1
+
+try:
+    while 1:
+        result = child.poll()
+        if result is not None:
+            break
+        if Backend=="LOCAL":
+            flush_file(outfile)
+            flush_file(errorfile)
+            time.sleep(0.3)
+        elif Backend=="BATCH":
+            heartbeatfile.write('.')
+            flush_file(heartbeatfile)
+            heartbeatsequence = ###HEARTBEATFREQUENCE###
+            time.sleep(int(heartbeatsequence))
+except Exception as x:
+    print('ERROR: %s'%str(x))
+finally:
+    pass
+    sys.stdout=sys.__stdout__
+    sys.stderr=sys.__stderr__
+
+flush_file(outfile)
+flush_file(errorfile)
+
+
+if Backend=="BATCH":
+    flush_file(sys.stdout)
+    flush_file(sys.stderr)
+    print("--- GANGA APPLICATION OUTPUT END ---", file=sys.stdout)
+
+try:
+    filefilter
+except:
+    filefilter = None
+
+from files import multi_glob, recursive_copy
+
+createOutputSandbox(outputpatterns,filefilter,sharedoutputpath)
+
+def printError(message):
+    print(message, file=errorfile)
+    flush_file(errorfile)
+
+def printInfo(message):
+    print(message, file=outfile)
+    flush_file(errorfile)
+
+
+###OUTPUTUPLOADSPOSTPROCESSING###
+
+if Backend=="BATCH":
+    print("--- GANGA APPLICATION ERROR END ---", file=sys.stderr)
+
+###OUTPUTSANDBOXPOSTPROCESSING###
+
+outfile.close()
+errorfile.close()
+
+
+###POSTEXECUTE###
+
+
+line='EXITCODE: ' + repr(result) + os.linesep
+line+='STOP: '+time.strftime('%a %b %d %H:%M:%S %Y',time.gmtime(time.time())) + os.linesep
+statusfile.writelines(line)
+statusfile.close()
+
+if Backend=="BATCH":
+    heartbeatfile.close()
+    os.unlink(heartbeatfilename)
```

Comment:
Again, unifying the monitoring, will avoid the "if" clause.

Reply:
`os.unlink` will delete the filepath. It is not required in the local backend. Running the `os.unlink` command means the jobs status is not update on the ipython terminal. I am not sure if it can be removed all together. Does batch need the heartbeatfile to be deleted?

Reply:
We can probably just leave it for the batch backend as well. How is it determined that the job has finished for the batch backend at the moment?

Reply:
I think through  [`__jobstatus__`](https://github.com/ganga-devs/ganga/blob/70d744b31ca3052335099d6fa4e51b3a3d9ba1ed/ganga/GangaCore/Lib/Batch/Batch.py#L222) file.